### ISSUE TITLE
JH 0.8.1 -> 0.9.2 upgrade `invalid_redirect_uri` error for running users

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/jupyterhub#2125]
## Z2JH deployment upgrade failure for running users
We recently released a new Z2JH Helm chart. Apparently, users with a running pod revisiting their server through the hub got to see the error below, if an upgrade from `v0.6` to `0.7.0` had been made in between visits of the running pod. The suspected change that could have caused this is that JupyterHub went from `0.8.1` to `0.9.2`, and the `--upgrade-db` flag (if I recall correctly) did some work when the JupyterHub started again.

```
{“error”: “invalid_redirect_uri”, “error_description”: “Invalid redirect URI”}
```

@minrk had a clue as to why this happened
> It means the db upgrade resulted in invalid oauth for the running users.
I think this might be something to fix in the jupyterhub alembic upgrades

**Comment by consideRatio** [comment]
No =/

I recall finding a bug with a key being lost because it wasn't properly read from the configmap or similar. @minrk do you know if a certain `openssl rand -hex 32` secret being reset could cause this issue? What could trigger such potential reset in a db upgrade as compared to just hub restarts? Hmmm may be a dead end investigating this but I put it out there anyhow for now.

**Comment by minrk** [comment]
It's not related to secrets. It's a change in the oauth client ids that isn't handled by the migration. This should be fixed by #2155. I don't know how to write an automated test for this, but I was able to test it manually by:

1. start jupyterhub 0.8
2. login, launch server
3. stop jupyterhub 0.8 (leaving servers running)
4. launch jupyterhub 0.9 with --upgrade-db
5. reload the page accessing the server
